### DISCUSSION TITLE
Writing custom query and getting it's results in antlers template

### DISCUSSION THREAD

**Discussion by imakeitworknet** [statamic/cms#8483] [Q&A] [answered]
Hi.
I'm trying to integrate a calendar to Statamic website. I have collection named Shows which has start_date and end_date fields.
Thing is that I need to get all dates (with title) when the event is happening - even if show is few days long.

If it was plain Laravel I could just write a custom function, add it to controller, and then display the results in some blade template. 

But where should I put this custom function in Statamic correctly, and how should I call it in Antlers then? I suppose query scope or filter won't give me what I want.

For example:
```
$data = Entry::query()
            ->where('collection', 'show')
            ...
            ->get();
foreach ($data...)
   $results[] = ...
endforeach

return $results;
```

Where this logic should be added correctly so it would not be eaten by updates, and how I can call it in antlers template?

Thanks!

**Comment by jasonvarga** [comment, marked as answer]
If you're using Antlers, the suggested way here would be to [create a "Tag"](https://statamic.dev/extending/tags) and perform your query in there.

Something like this:

```php
public function index()
{
  return Entry::query()->etc()->get();
}
```

```antlers
{{ your_tag }}
  {{ title }}, {{ etc }}
{{ /your_tag }}
```

**Reply by imakeitworknet** [reply]
Thanks, it works!
